### ISSUE TITLE
zone.js can't patch XHR requests on IE11

### ISSUE THREAD

**Issue by kmaraz** [angular/angular#38561]
# 🐞 bug report

### Affected Package
The issue is caused by package zone.js@0.11.1

### Is this a regression?

Yes, the previous version in which this bug was not present was: 0.10.3

### Description

This error occurs on IE11 when zone.js tries to patch any XHR requests.

## 🔥 Exception or Error
Error message is 
<pre><code>
"Invalid calling object"
</code></pre>

Stacktrace
<pre><code>
at scheduleTask (https://mypage/static/npm.zone.js.4a7cc42837185b0e9a27.js:3359:9)
at scheduleTask (https://mypage/static/npm.zone.js.4a7cc42837185b0e9a27.js:503:15)
at onScheduleTask (https://mypage/static/npm.zone.js.4a7cc42837185b0e9a27.js:390:9)
at scheduleTask (https://mypage/static/npm.zone.js.4a7cc42837185b0e9a27.js:498:13)
</code></pre>

![image](https://user-images.githubusercontent.com/6079654/91032365-25fecc80-e602-11ea-9162-1fa2903d19b5.png)
![image](https://user-images.githubusercontent.com/6079654/91032426-36af4280-e602-11ea-95af-fa4c9f773a41.png)
![image](https://user-images.githubusercontent.com/6079654/91032458-43339b00-e602-11ea-84a4-07c4dbdc0616.png)

## 🌍  Your Environment

**Angular Version:**
I am using custom Webpack configuration. All the packages are current latest version 10.0.11.

**Comment by BdDragos** [comment]
I also have the same error, reverting back to 0.10.3 from 0.11.1 fixed it. It was caught by my error interceptor:

`  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
    return next.handle(request).pipe(
      catchError((error: HttpErrorResponse) => {
        let errorMessage = '';
        let displayedMessage = '';
        if (error.error instanceof ErrorEvent) {
          // client-side error
          errorMessage = `Error: ${error.error.message}`;
          displayedMessage = 'Request to server failed - Client-side';
        } else {
          // server-side error
          errorMessage = `Error Code: ${error.status}\nMessage: ${error.message}`;
          displayedMessage = 'Request to server failed - Code ' + error.status;
        }
        this.loadingSpinnerService.closeAll();
        this.toastService.show({ text: displayedMessage, type: 'error' });
        return throwError(errorMessage);
      })
    );
  }`

**Comment by JiaLiPassion** [comment]
@kmaraz, I can not reproduce this issue, could you provide a reproduce repo? thanks.

**Comment by kmaraz** [comment]
@JiaLiPassion here: https://github.com/kmaraz/zonejs-demo
Hope it helps. I am able to reproduce this with this minimal demo.

**Comment by kmaraz** [comment]
Maybe related: Also I am using `blacklist.ts` to create new Zone which doesn't work anymore:
```
const BLACKLISTED_ZONE_EVENTS: string[] = [
  // Global events
  'addEventListener:mouseenter',
  'addEventListener:mouseleave',
  'addEventListener:mouseout',
  'addEventListener:mouseover',
  'addEventListener:mousewheel',
  'addEventListener:scroll',
  'addEventListener:wheel',
  'addEventListener:resize',
  'addEventListener:orientationchange',
  // Chart.js
  'addEventListener:pointermove',
  // Animations
  'requestAnimationFrame',
  // Websockets
  'WebSocket.addEventListener:close',
  'WebSocket.addEventListener:error',
  'WebSocket.addEventListener:open',
  'WebSocket.addEventListener:message'
];

export const blacklistZone = Zone.current.fork({
  name: 'blacklist',
  onScheduleTask: (delegate: ZoneDelegate, _current: Zone, target: Zone, task: Task): Task => {
    if (
      // Block events
      task.type === 'eventTask' && BLACKLISTED_ZONE_EVENTS.some((name) => task.source.includes(name))
    ) {
      task.cancelScheduleRequest();
      return Zone.root.scheduleTask(task);
    }
    return delegate.scheduleTask(target, task);
  }
});
```
Because `Zone`, `ZoneDelegate` and `Task` are unrecognized.

**Comment by JiaLiPassion** [comment]
@kmaraz , the repo https://github.com/kmaraz/zonejs-demo only contains some config files, could you post a workable repo?

**Comment by kmaraz** [comment]
@JiaLiPassion sorry, it is updated now.

**Comment by JiaLiPassion** [comment]
@kmaraz, thanks for the sample, this is not a bug since zone.js 0.11.1 is a major update with breaking change since from 0.11.1, zone.js change the `dist` bundle format to `Angular Package Format`, and your project is not generated with `Angular CLI`, so it will not work in IE 11 without update the webpack config, since currently Angular 10 is still using zone.js 0.10.3, so please don't update to zone.js 0.11.1 if you are not using Angular CLI. I will update the document to describe about this.

And for now, the walk around is, update `app.module.ts` to update 
`import zone.js'
to 
`import 'zone.js/bundles/zone.umd';`

Thanks for reporting the issue.

**Comment by sir-captainmorgan21** [comment]
@JiaLiPassion why would `app.module.ts` be importing from zone.js? Do you mean `polyfills.ts`?

**Comment by JiaLiPassion** [comment]
@sir-captainmorgan21 , sure, if the project is created by `Angular CLI`, zone.js is imported in `polyfills.ts`, and also if the project is `Angular CLI` project, after updating to zone.js 0.11.x, it will still work.

The issue here https://github.com/kmaraz/zonejs-demo is not an `Angular CLI` project, the zone.js is imported in `app.module.ts` and the syntax there is `import 'zone.js';`, and this syntax become a breaking change since zone.js 0.11.x now is Angular package format, and import 'zone.js' will load zone bundles for evergreen browsers and not loading packages for legacy browsers such as IE

**Comment by kmaraz** [comment]
Off topic:
>  since zone.js 0.11.1 is a major update with breaking change
It would be nice if zone.js would follow semver. How it is possible that major version didn't change but it is a breaking change. I'd never guessed that.

**Comment by JiaLiPassion** [comment]
@kmaraz, yeah, you are right, I understand your point, but since the major version of `zone.js` is still `0` and it may keep that way for a long time, so now we use the minor version to identify the breaking changes.

**Comment by JiaLiPassion** [comment]
@kmaraz, I just made another PR to fix this issue, zone.js 0.11 should have full backward compatibility, please wait for the next release 0.11.2, so you don't need to update your code.

**Comment by pxw84** [comment]
Can I use latest zone.js with Angular 8 ? I have the same issue with `Invalid calling object` and I'm experimenting with different zone.js versions.

**Comment by JiaLiPassion** [comment]
@pxw84 , sure, you can use the lastest version with `Angular 8`, but currently the `0.11.1` has a unintended breaking change, the issue has been fixed, please wait for `0.11.2`, it will be release very soon.
